lib/std/crypto/25519/scalar.zig

```diff
@@ -0,0 +1,177 @@
+const std = @import("std");
+const mem = std.mem;
+
+const field_size = [32]u8{
+    0xed, 0xd3, 0xf5, 0x5c, 0x1a, 0x63, 0x12, 0x58, 0xd6, 0x9c, 0xf7, 0xa2, 0xde, 0xf9, 0xde, 0x14, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10, // 2^252+27742317777372353535851937790883648493
+};
+
+const ScalarExpanded = struct {
+    limbs: [64]i64 = [_]i64{0} ** 64,
+
+    fn fromBytes(s: [32]u8) ScalarExpanded {
+        var limbs: [64]i64 = undefined;
+        for (s) |x, idx| {
+            limbs[idx] = @as(i64, x);
```

Comment:
no need for `@as` here, the coercion will happen implicitly.

Reply:
IMHO, it really helps understand what the code does and that it was intentional. Reading `a[i] = x` without paying attention to the context feels like a simple copy.